lib/screen/chat/chat_thread_page.dart

```diff
@@ -234,19 +164,30 @@ class _ChatThreadPageState extends State<ChatThreadPage> {
             status: types.Status.delivered,
           ));
     }
-    setState(() {});
+    if (mounted) {
+      setState(() {});
```

Comment:
please `setState` for only the changed fields, unless it will rebuild all.
https://dcm.dev/docs/rules/flutter/avoid-empty-setstate/

Reply:
in this case, it it rebuild the chat, so basically rebuild all but the navigation bar.
But the reason i put it that way is if i wrap where the field change, i have to wrap 3 places and it sometime would rebuild twice.
What do you think about that trade off?

Reply:
plus, i have to check mounted, because this method is called in a socket channel event, so it may be unmounted